### PR TITLE
tests: improve Code Coverage of ConfigurationService (SDKCF-6388)

### PR DESCRIPTION
**Pull request by ts-robih-robihamanto**
# Description
Improved coverage of `ConfigurationService` to **98.9%**
This else line cannot failed, because it's already checked in the previous line if it's link is a valid URL:
```swift
guard let url = urlComponents?.url else {
    return .failure(RequestError.urlIsNil)
}
```

## Links
SDKCF-6388

# Checklist
- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have added to the [changelog](../blob/master/CHANGELOG.md#Unreleased)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys, and internal links
- [ ] I ran `fastlane ci` without errors
- [x] All project file changes are replicated in `SampleSPM/SampleSPM.xcodeproj` project

### CODE REVIEW

File: Tests/Tests/ConfigurationServiceSpec.swift [hunk 1]
```diff
@@ -254,6 +254,15 @@ class ConfigurationServiceSpec: QuickSpec {
                 }
             }
 
+            context("when building request body") {
+
+                it("will return RequestError.bodyIsNil") {
```
Comment (imaginaris) [inline Tests/Tests/ConfigurationServiceSpec.swift]: please add test cases that check if the same is returned when you pass a non-nil parameter in `buildHttpBody()` so it's clear that `bodyIsNil` is always returned no matter what parameter is passed

**Review by imaginaris — CHANGES_REQUESTED** [review]
The `assertionFailure()` in `buildRequestHeader()` can be tested by setting subscription id to nil and using `to(throwAssertion())` predicate. It won't crash the unit test run

**Comment by ts-robih-robihamanto** [comment]
> The `assertionFailure()` in `buildRequestHeader()` can be tested by setting subscription id to nil and using `to(throwAssertion())` predicate. It won't crash the unit test run

I added the case for throwAssertion here [0726cfa](https://github.com/rakutentech/ios-inappmessaging/pull/261/commits/62da5bc73d4b56f83df4c5bd4f4234dd20726cfa).

**Review by imaginaris — APPROVED** [review]
👍
